### PR TITLE
Feature/auto shell exec

### PR DESCRIPTION
**Pull request by rishabh-ti**
Add/delete components right from the UI.

https://user-images.githubusercontent.com/110235735/203854429-726389ce-f97c-480a-8690-c9a21b736eae.mov

### CODE REVIEW

**Comment by vickybiswas** [comment]
The video looks Good, Please add a loader since deleting takes time.

**Comment by vickybiswas** [comment]
Why did we choose this mechanism above exec or spawn?

File: src/pages/api/add-component.js [hunk 1]
```diff
@@ -3,10 +3,10 @@ import shell from 'shelljs'
 
 export default async function handler(req, res) {
   try {
-    const response = await shellExec('bit --version')
-    const response2 = shell.exec('bit --version')
+    // const response = await shellExec('bit --version')
+    const response2 = shell.exec(`bit create tiui-react-oc ${req.body.path}`)
```
Comment (vickybiswas) [inline src/pages/api/add-component.js]: This is potentially dangerous Validate and Sanitize before passing to exec

**Comment by vickybiswas** [comment]
It will be good to add a pop confirmation to avoid mistakes while deletion

**Comment by rishabh-ti** [comment]
> The video looks Good, Please add a loader since deleting takes time.

Merged already, please check this video for reference: https://github.com/trilogy-group/tu2k22-openchakra/pull/35#issue-1464801760

**Comment by rishabh-ti** [comment]
> Why did we choose this mechanism above exec or spawn?

[Shelljs](https://www.npmjs.com/package/shelljs), the package used, internally uses child_process_exec through NodeJs. Using a package as it is OS agnostic and would help when the user wants to work locally on any OS.

**Comment by rishabh-ti** [comment]
> Added Delete confirmation dialog

Added: https://github.com/trilogy-group/tu2k22-openchakra/pull/32/commits/5902b1c51bb9f12c641d9101ada10534fccf47c2

File: src/pages/api/add-component.js (hunk 1 above)
Comment (rishabh-ti) [reply src/pages/api/add-component.js]: Added: https://github.com/trilogy-group/tu2k22-openchakra/pull/32/commits/26d614ce1206e8782af2c67329728729610d0285

**Pull request merged**